samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -129,4 +129,8 @@ public ApplicationStatus status(StreamApplication app) {
     }
   }
 
+  @Override
+  public void waitForFinish() {
```

Comment:
Maybe just move this method to LocalApplicationRunner instead of the AbstractApplicationRunner? No other runners implement this.